src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -221,7 +221,7 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
             throw new RuntimeException(sprintf('Invalid handler service "%s": type-hint of argument "$%s" in method "%s::__invoke()" must be a class , "%s" given.', $serviceId, $parameters[0]->getName(), $handlerClass->getName(), $type));
         }
 
-        return [(string) $parameters[0]->getType()];
+        return [$parameters[0]->getType()->getName()];
```

Comment:
it seems `getName` only exists on ReflectionNamedType https://www.php.net/manual/en/reflectionnamedtype.getname.php and https://php.net/manual/en/reflectionparameter.gettype.php says it only returns ReflectionType. So getName would not be safe to call here.
But that looks like the php doc is wrong. When the param does not have a type `getType` returns null. So getType always returns ReflectionNamedType|null.

Reply:
That's because currently, ReflectionNamedType is the only subtype of ReflectionType
ref: https://github.com/phpstan/phpstan/issues/1133#issuecomment-401194968